packages/cli/src/controller/migrate/manifest/migrate-manifest.controller.ts

```diff
@@ -0,0 +1,121 @@
+// Copyright 2020-2024 SubQuery Pte Ltd authors & contributors
+// SPDX-License-Identifier: GPL-3.0
+
+import fs from 'fs';
+import path from 'path';
+import {NETWORK_FAMILY} from '@subql/common';
+import {CommonSubqueryProject} from '@subql/types-core';
+import {upperFirst} from 'lodash';
+import YAML from 'yaml';
+import {findRunnerByNetworkFamily, prepareDirPath, renderTemplate} from '../../../utils';
+import {TemplateKind} from '../../codegen-controller';
+import {graphToSubqlNetworkFamily, networkDsConverters, networkTemplateConverters} from '../constants';
+import {getChainIdByNetworkName} from '../migrate-controller';
+import {MigrateDatasourceKind, SubgraphDataSource, SubgraphProject, SubgraphTemplate, ChainInfo} from '../types';
+
+const PROJECT_TEMPLATE_PATH = path.resolve(__dirname, '../../../template/project.ts.ejs');
+
+export function readSubgraphManifest(inputPath: string): SubgraphProject {
+  if (!fs.existsSync(inputPath)) {
```

Comment:
I think rather than check this, we can just handle the `ENOENT` error when calling `readFileSync`